geode-core/src/main/java/org/apache/geode/internal/cache/wan/AbstractGatewaySender.java

```diff
@@ -126,6 +127,8 @@
 
   protected boolean groupTransactionEvents;
 
+  protected volatile boolean isStopping = false;
```

Comment:
can we have a better name, such as isStoppingForTX?

Reply:
I would prefer to leave it as it is. The variable is just indicating that we are in the process of stopping the gateway sender.
I have changed the `AbstractGatewaySender.preStop()` method to make this more clear.